### PR TITLE
Update spec to v4.0.0

### PR DESCRIPTION
**Pull request by eabadjiev**
Changes inc:
- verion: `4.0.0`
- tag: `v4.0.0`
- tvOS UI sources

Verified with `pod spec lint ReactiveCocoa.podspec`:

```
ReactiveCocoa.podspec passed validation.
```

### CODE REVIEW

**Comment by ashfurrow** [comment]
Looks 👍 to me. Will push to trunk later when I get to a computer.

**Comment by ashfurrow** [comment]
Alrighty! I've got that pushed up and you should be good to go: https://raw.githubusercontent.com/CocoaPods/Specs/c9c108fa860feccce38126fc946085a4938d31b0/Specs/ReactiveCocoa/4.0.0/ReactiveCocoa.podspec.json

Let us know if you have any issues, and thanks again!

**Comment by eabadjiev** [comment]
Works for me 👍, thanks for the quick reaction @ashfurrow. 

If someone removes the 4.0.x-alpha\* versions, I will be very happy :)

**Comment by orta** [comment]
Good timing, I did _just_ ship a working version of `pod trunk delete` on the server - you'll need to use a CP 1.0 beta

**Comment by ashfurrow** [comment]
Is it a good idea to delete them? Some people may be relying on them, and the resolver should ignore them unless they're specifically specified.

**Comment by orta** [comment]
I'm not taking responsibility for that choice. But this is going to be mighty confusing for people.

```
4.0.1-alpha-1
4.0.1-alpha-3
4.0.2-alpha-1
4.0.2-alpha-3
4.0.3-alpha-1
4.0.3-alpha-3
4.0.4-alpha-1
4.0.4-alpha-4
```

**Comment by eabadjiev** [comment]
My 2 cents are:

1) CocoaPods will show the package as outdated, which in long turn will do more harm than good. Try `pod outdated`

```
Updating spec repo `master`
....
Analyzing dependencies
The following pod updates are available:
- ReactiveCocoa 4.0.0 -> 4.0.0 (latest version 4.0.4-alpha-4)
```

2) These are alpha builds which you must manually / explicitly alter to be able to submit with. Developers should be aware and very keen to get latest and greatest. On top the API breaking changes are very minimal.

**Comment by ashfurrow** [comment]
Whatevs works ¯\_(ツ)_/¯

File: ReactiveCocoa.podspec [hunk 1]
```diff
@@ -60,7 +60,10 @@ Pod::Spec.new do |s|
     ]
     ss.osx.framework = 'AppKit'
 
-    ss.tvos.source_files = '**/ReactiveCocoa.h'
+    ss.tvos.source_files = [
+      '**/ReactiveCocoa.h',
+      'ReactiveCocoa/**/*{UIButton,UICollectionReusableView,UIControl,UIGestureRecognizer,UISegmentedControl,UITableViewCell,UITableViewHeaderFooterView,UIText}*'
```
Comment (NachoSoto) [inline ReactiveCocoa.podspec:65]: Yikes, this information is repeated in 4 places now.

Comment (ashfurrow) [inline ReactiveCocoa.podspec:65]: Yeah, worth abstracting somewhere reusable. Want to open an issue?

Comment (NachoSoto) [inline ReactiveCocoa.podspec:65]: What are some examples of this? 2 of the other 3 places are in the project file.

Comment (eabadjiev) [inline ReactiveCocoa.podspec:65]: Please, keep in mind that not all iOS UI classes are available for tvOS.

Comment (ashfurrow) [inline ReactiveCocoa.podspec:65]: Opened https://github.com/mhuusko5/Podspecs/issues/7 and will try to get to it today.

**Pull request merged**